tough/src/editor/signed.rs

```diff
@@ -701,25 +791,27 @@ trait TargetsWalker {
         // unique; if we're not, then there could be a target from another repo with the same name
         // but different checksum.  We can't assume such conflicts are OK, so we fail.
         if !self.consistent_snapshot() {
-            // Use DigestAdapter to get a streaming checksum of the file without needing to hold
-            // its contents.
-            let f = fs::File::open(&dest).context(error::FileOpenSnafu { path: &dest })?;
-            let mut reader = DigestAdapter::sha256(
-                Box::new(f),
-                &repo_target.hashes.sha256,
-                Url::from_file_path(&dest)
-                    .ok() // dump unhelpful `()` error
-                    .context(error::FileUrlSnafu { path: &dest })?,
-            );
-            let mut dev_null = std::io::sink();
+            let url = Url::from_file_path(&dest)
+                .ok() // dump unhelpful `()` error
+                .context(error::FileUrlSnafu { path: &dest })?;
+
+            let stream = FilesystemTransport
+                .fetch(url.clone())
+                .await
+                .with_context(|_| error::TransportSnafu { url: url.clone() })?;
+            let stream = DigestAdapter::sha256(stream, &repo_target.hashes.sha256, url.clone());
+
             // The act of reading with the DigestAdapter verifies the checksum, assuming the read
             // succeeds.
-            std::io::copy(&mut reader, &mut dev_null)
-                .context(error::FileReadSnafu { path: &dest })?;
+            stream
+                .try_for_each(|_| ready(Ok(())))
+                .await
+                .with_context(|_| error::TransportSnafu { url: url.clone() })?;
```

Comment:
I think `.clone()` is not necessary anymore at this point.

Reply:
clippy why have you forsaken me?!